Reyes_JohnRuiz/script.js

```diff
@@ -0,0 +1,74 @@
+document.addEventListener("DOMContentLoaded", function () {
+  const commentNameInput = document.getElementById("comment-name");
+  const commentTextInput = document.getElementById("comment-text");
+  const submitCommentButton = document.getElementById("submit-comment");
+  const commentsContainer = document.getElementById("comments-container");
+  const sortAscButton = document.getElementById("sort-asc");
+  const sortDescButton = document.getElementById("sort-desc");
+
+  let comments = [];
+
+  function checkCommentValidity() {
+    const nameValue = commentNameInput.value.trim();
+    const textValue = commentTextInput.value.trim();
+    submitCommentButton.disabled = !(nameValue && textValue);
+  }
+
+  checkCommentValidity();
+
+  commentNameInput.addEventListener("input", checkCommentValidity);
+  commentTextInput.addEventListener("input", checkCommentValidity);
+
+  submitCommentButton.addEventListener("click", (e) => {
+    e.preventDefault();
+    const name = commentNameInput.value.trim();
+    const text = commentTextInput.value.trim();
+
+    if (!name || !text) {
+      return;
+    }
+
+    addComment(name, text, new Date());
+    commentNameInput.value = "";
+    commentTextInput.value = "";
+    checkCommentValidity();
+  });
+
+  function addComment(name, text, date) {
+    const commentElement = document.createElement("p");
+    commentElement.innerHTML = `Name: ${name}<br>Comment: 
+        ${text}<br><small>${date.toLocaleString()}</small>`;
+    commentsContainer.appendChild(commentElement);
+
+    comments.push({ name, text, date });
+  }
+
+  sortAscButton.addEventListener("click", () => {
```

Comment:
Do not use arrow function as a method of an object. You can use the sample code and adjust it to your intended use.
```suggestion
  sortAscButton.addEventListener("click", function() => {
```